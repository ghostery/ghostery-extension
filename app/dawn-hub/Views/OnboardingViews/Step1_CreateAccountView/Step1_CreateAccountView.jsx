/**
 * Browser Create Account View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Fragment, useState } from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import Step1_LogInForm from '../Step1_LogInForm';
import Step1_CreateAccountForm from '../Step1_CreateAccountForm';
import globals from '../../../../../src/classes/Globals';
import {
	LOGIN,
	ONBOARDING,
	SIGN_IN_SUCCESSFUL,
	CREATE_ACCOUNT_SUCCESSFUL,
	SKIP_ACCOUNT_CREATION
} from '../../OnboardingView/OnboardingConstants';

const SIGN_IN = 'SIGN_IN';
const CREATE_ACCOUNT = 'CREATE_ACCOUNT';

const faqList = [
	{
		icon: 'ghosty-shield.svg',
		label: t('ghostery_dawn_onboarding_private_by_design'),
		description: t('ghostery_dawn_onboarding_private_by_design_description'),
	},
	{
		icon: 'ghosty-box.svg',
		label: t('ghostery_dawn_onboarding_can_i_remove_my_account'),
		description: t('ghostery_dawn_onboarding_can_i_remove_my_account_description'),
	}
];

const renderFAQListItem = (icon, label, description) => (
	<div key={label} className="Step1_CreateAccountView__faqItemContainer row">
		<div className="Step1_CreateAccountView__faqIconContainer columns small-12 medium-10 large-2">
			<img className="Step1_CreateAccountView__faqIcon" src={`/app/images/hub/browser-create-account-view/${icon}`} />
		</div>
		<div className="Step1_CreateAccountView__faqItemTextContainer columns small-12 medium-10 large-10">
			<div className="Step1_CreateAccountView__faqItemLabel">{label}</div>
			<div className="Step1_CreateAccountView__faqItemDescription">{description}</div>
		</div>
	</div>
);

/**
 * A Functional React component for rendering the Browser Create Account View
 * @return {JSX} JSX for rendering the Browser Create Account View of the Dawn Hub app
 * @memberof DawnHubViews
 */
const Step1_CreateAccountView = (props) => {
	const { actions, step, user } = props;
	const { setSetupStep, setToast } = actions;
	const email = user && user.email;

	const [view, setView] = useState(CREATE_ACCOUNT);

	const handleSkipButton = (dawn_setup_number) => {
		setSetupStep({
			setup_step: LOGIN,
			dawn_setup_number,
			origin: ONBOARDING,
		});
		setToast({
			toastMessage: '',
			toastClass: ''
		});
	};

	const handleNextOnSelectPlanStep = () => {
		const { prev } = props;

		setSetupStep({
			setup_step: LOGIN,
			dawn_setup_number: CREATE_ACCOUNT_SUCCESSFUL,
			origin: ONBOARDING,
		});

		setToast({
			toastMessage: '',
			toastClass: ''
		});

		prev();
	};

	const renderSkipLink = () => (
		<div className="row align-center-middle">
			<div className="columns small-10 medium-6" />
			<div className="columns small-10 medium-6">
				<NavLink className="Step1_CreateAccountView__skip" to="/onboarding/2" onClick={() => handleSkipButton(SKIP_ACCOUNT_CREATION)}>
					<span>{t('ghostery_dawn_onboarding_skip')}</span>
				</NavLink>
			</div>
		</div>
	);

	const subtitle = (step === LOGIN)
		? t('ghostery_dawn_onboarding_sync_settings')
		: t('ghostery_dawn_onboarding_create_account_for_trial');

	return (user ? (
		<div className="Step1_CreateAccountView__alreadySignedIn">
			<div className="Step1_CreateAccountView__title">{t('ghostery_dawn_onboarding_you_are_signed_in_as')}</div>
			<div className="Step1_CreateAccountView__email">{email}</div>
			<div className="Step1_CreateAccountView__ctaButtonContainer">
				{step === LOGIN && (
					<NavLink className="Step1_CreateAccountView__ctaButton" to="/onboarding/2" onClick={() => handleSkipButton(SIGN_IN_SUCCESSFUL)}>
						<span>{t('next')}</span>
					</NavLink>
				)}
				{step !== LOGIN && (
					<div className="Step1_CreateAccountView__ctaButton" onClick={() => handleNextOnSelectPlanStep()}>
						<span>{t('next')}</span>
					</div>
				)}
			</div>
		</div>
	) : (
		<div className="Step1_CreateAccountView">
			{step !== LOGIN && (
				<div className="CreateAccountView__relativeContainer">
					<div className="CreateAccountView__backContainer">
						<span className="CreateAccountView__caret left" />
						<NavLink to="/onboarding/3">
							<span className="CreateAccountView__back">{t('ghostery_dawn_onboarding_back_to_search_selection')}</span>
						</NavLink>
					</div>
				</div>
			)}
			{view === CREATE_ACCOUNT && (
				<div className="Step1_CreateAccountView__title">{t('ghostery_dawn_onboarding_create_a_ghostery_account')}</div>
			)}
			{view === SIGN_IN && (
				<div className="Step1_CreateAccountView__title">{t('sign_in')}</div>
			)}
			<div className="Step1_CreateAccountView__subtitle">{subtitle}</div>
			<div className="row align-center-middle">
				{view === CREATE_ACCOUNT && (
					<div className="Step1_CreateAccountView__alreadyHaveAccount columns small-12" onClick={() => setView(SIGN_IN)}>{t('ghostery_dawn_onboarding_already_have_account')}</div>
				)}
				{view === SIGN_IN && (
					<div className="Step1_CreateAccountView__alreadyHaveAccount columns small-12" onClick={() => setView(CREATE_ACCOUNT)}>{t('ghostery_dawn_onboarding_create_an_account')}</div>
				)}
			</div>
			{view === CREATE_ACCOUNT ? (
				<Fragment>
					{/* eslint-disable-next-line react/jsx-pascal-case */}
					<Step1_CreateAccountForm />
					{(step === LOGIN) && renderSkipLink()}
					<div className="Step1_CreateAccountView__FAQContainer">
						{faqList.map(item => renderFAQListItem(item.icon, item.label, item.description))}
					</div>
					<div className="row">
						<a className="Step1_CreateAccountView__privacyPolicyLink columns small-12 medium-10 medium-offset-1 large-8 large-offset-3" href={`${globals.GHOSTERY_BASE_URL}/about-ghostery/ghostery-plans-and-products-privacy-policy/`} target="_blank" rel="noreferrer">
							{t('ghostery_dawn_onboarding_visit_our_privacy_policy')}
						</a>
					</div>
				</Fragment>
			) : (
				<Fragment>
					{/* eslint-disable-next-line react/jsx-pascal-case */}
					<Step1_LogInForm />
					{(step === LOGIN) && renderSkipLink()}
				</Fragment>
			)}
		</div>
	));
};

// PropTypes ensure we pass required props of the correct type
Step1_CreateAccountView.propTypes = {
	user: PropTypes.shape({
		email: PropTypes.string,
	}),
};

// Default props used in the Plus View
Step1_CreateAccountView.defaultProps = {
	user: {
		email: ''
	},
};

export default Step1_CreateAccountView;
