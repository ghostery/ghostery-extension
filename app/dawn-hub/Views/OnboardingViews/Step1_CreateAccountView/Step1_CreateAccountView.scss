/**
 * Step1_CreateAccountView Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

$color-create-account-form-error-red: #e74055;

// Browser Create Account View
.Step1_CreateAccountView {
	padding-top: 80px;
	padding-bottom: 40px;
}

.CreateAccountView__relativeContainer {
  position: relative;
  left: 30px;
  top: -70px;
}

.CreateAccountView__caret.left {
  margin: 7px auto 0 auto;
  height: 10px;
  width: 10px;
  border-left: 2px solid $tundora;
  border-top: 2px solid $tundora;
  cursor: pointer;
  transform: rotate(-45deg);
}

.CreateAccountView__backContainer {
  position: absolute;
  display: flex;
  margin-top: 60px;
  @include breakpoint(xlarge down) {
	margin-top: 22px;
  }
  .CreateAccountView__back {
	margin-top: 8px;
	font-size: 16px;
	color: $tundora;
	text-decoration: underline;
  }
}

.Step1_CreateAccountView__alreadySignedIn {
	margin-top: 245px;
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}
.Step1_CreateAccountView__email {
	font-size: 24px;
	color: $ghosty-blue;
}
.Step1_CreateAccountView__ctaButtonContainer {
	display: flex;
	justify-content: center;

	.Step1_CreateAccountView__ctaButton {
		margin: 48px auto 0 auto;
		min-height: 44px;
		width: 162px;
		padding: 7.7px 14px;
		line-height: 22px;
		background: linear-gradient(
			45deg,
			#ff7e74 50%,
			#00aef0
		);
		background-size: 200% 100%;
		background-position: 100% 50%;
		transition: 0.25s all;
		border: none;
		&:hover {
			background-position: 0% 50%;
			transition: 0.25s all;
		}
		color: #FFF;
		font-size: 14.1px;
		font-weight: 700;
		border-radius: 3.5px;
		text-align: center;
		line-height: 2.05;
		cursor: pointer;
	}
}
.Step1_CreateAccountView__title {
	display: flex;
	justify-content: center;
	font-size: 24px;
	font-weight: 600;
	@include breakpoint(small down) {
		margin: auto;
		max-width: 237px;
		text-align: center;
	}
}
.Step1_CreateAccountView__subtitle {
	margin: 10px auto 0 auto;
	display: flex;
	justify-content: center;
	font-size: 18px;
	text-align: center;
  	max-width: 600px;
	@include breakpoint(small down) {
		margin: auto;
		max-width: 237px;
		text-align: center;
		line-height: 2.33;
	}
}
.Step1_CreateAccountView__alreadyHaveAccount {
	margin-top: 38px;
	font-size: 16px;
	color: $ghosty-blue;
	text-decoration: underline;
	cursor: pointer;
	@include breakpoint(small down) {
		margin: 38px auto 0 auto;
		text-align: center;

		&:nth-child(2) {
			margin: 0;
		}
	}
}
.Step1_CreateAccountView--addMarginSide {
	margin-left: 10px;
	margin-right: 10px;
}
.Step1_CreateAccountView--addMarginBottom {
	margin-bottom: 20px;
}
.Step1_CreateAccountView__headerImage {
	max-width: 180px;
}
.Step1_CreateAccountView__headerTitle h3 {
	font-size: 24px;
	line-height: 1.75;
	color: #4a4a4a;
	margin: 0 25px;
}
.Step1_CreateAccountView__inputBox.error {
	// Foundation Overrides
	box-shadow: none;
	border-color: 2px solid $color-create-account-form-error-red;
}
.Step1_CreateAccountView__inputBox:focus {
	// Foundation Overrides
	box-shadow: none;
	border-color: $color-create-account-form-error-red;
}
.Step1_CreateAccountView__inputError {
	font-size: 12;
	line-height: 14px;
	color: $color-create-account-form-error-red;
	margin-bottom: 13px;
}
.Step1_CreateAccountView__link {
	font-size: 14px;
	line-height: 30px;
}
.Step1_CreateAccountView__termsStatement {
	font-size: 12px;
}
.Step1_CreateAccountView__button {
	min-width: 180px;
}
.Step1_CreateAccountView__inputLabel {
	flex-grow: 1; // ensure left-justify alignment in single column layout
	padding-top: 10px; // align with checkbox top
}
.Step1_CreateAccountView__inputLabel.error{
	color: $color-create-account-form-error-red;
}
.Step1_CreateAccountView__skip {
	margin-top: -37px;
	float: right;
	font-size: 16px;
	color: $ghosty-blue;
	text-decoration: underline;
	cursor: pointer;
	@include breakpoint(small down) {
		margin: 39px auto 0 auto;
		display: flex;
		justify-content: center;
		float: none;
	}
}
.Step1_CreateAccountView__learnMoreContainer {
	margin: 25px auto 0 auto;
	width: 251px;
	text-align: center;
}
.Step1_CreateAccountView__learnMore {
	font-size: 16px;
	color: $ghosty-blue;
	text-decoration: underline;
	cursor: pointer;
}
.Step1_CreateAccountView__FAQContainer {
	margin-top: 48px;
	padding-top: 20px;
	@include breakpoint(small down) {
		margin-top: 42px;
		text-align: center;
	}
	@include breakpoint(large up) {
		width: 108%;
		margin-left: -4%;
	}
}
.Step1_CreateAccountView__faqItemContainer {
	margin-bottom: 30px;
}
.Step1_CreateAccountView__faqIcon {
	height: 71px;
	width: 71px;
	min-height: 71px;
	min-width: 71px;
	margin-bottom: 10px;
}
.Step1_CreateAccountView__faqItemTextContainer {
	display: flex;
	flex-direction: column;
}
.Step1_CreateAccountView__faqItemLabel {
	font-size: 18px;
	font-weight: 600;
	color: #4a4a4a;
}
.Step1_CreateAccountView__faqItemDescription {
	margin-top: 10px;
	max-width: 607px;
	font-size: 16px;
	color: #4a4a4a;
}
.Step1_CreateAccountView__faqItemDescription {
	margin-top: 10px;
	max-width: 607px;
	font-size: 16px;
	color: #4a4a4a;
	@include breakpoint(small down) {
		margin: 16px auto 0 auto;
		width: 245px;
	}
}
.Step1_CreateAccountView__privacyPolicyLink {
	font-size: 16px;
	color: $ghosty-blue;
	text-decoration: underline;
	cursor: pointer;
	@include breakpoint(small down) {
		margin: auto;
		max-width: 271px;
		text-align: center;
	}
}
@media only screen and (max-width: 740px)  {
	.Step1_CreateAccountView {
		margin-bottom: 40px;
		padding-top: 80px;
	}
	.Step1_CreateAccountView__header {
		flex-direction: column;
	}
	.Step1_CreateAccountView__headerTitle {
		padding-top: 30px;
		text-align: center;
	}
	.Step1_CreateAccountView__submit {
		display: flex;
		justify-content: center;
	}
	.Step1_CreateAccountView__linkContainer {
		text-align: center;
	}
}
@media only screen and (max-width: 500px)  {
	.Step1_CreateAccountView__headerTitle h3 {
		font-size: 20px;
		margin: 0 10px;
	}
}
