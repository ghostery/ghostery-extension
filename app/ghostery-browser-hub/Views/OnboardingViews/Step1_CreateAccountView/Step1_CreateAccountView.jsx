/**
 * Browser Create Account View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Fragment, useRef, useState } from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import Step1_LogInForm from '../Step1_LogInForm';
import Step1_CreateAccountForm from '../Step1_CreateAccountForm';
import globals from '../../../../../src/classes/Globals';
import { LOGIN, ONBOARDING } from '../../OnboardingView/OnboardingConstants';

const SIGN_IN = 'SIGN_IN';
const CREATE_ACCOUNT = 'CREATE_ACCOUNT';

const faqList = [
	{
		icon: 'ghosty-shield.svg',
		label: t('ghostery_browser_hub_onboarding_private_by_design'),
		description: t('ghostery_browser_hub_onboarding_private_by_design_description'),
	},
	{
		icon: 'ghosty-box.svg',
		label: t('ghostery_browser_hub_onboarding_can_i_remove_my_account'),
		description: t('ghostery_browser_hub_onboarding_can_i_remove_my_account_description'),
	}
];

const renderFAQListItem = (icon, label, description) => (
	<div className="Step1_CreateAccountView__faqItemContainer row">
		<div className="Step1_CreateAccountView__faqIconContainer columns small-12 medium-10 medium-offset-1 large-2 large-offset-1">
			<img className="Step1_CreateAccountView__faqIcon" src={`/app/images/hub/browser-create-account-view/${icon}`} />
		</div>
		<div className="Step1_CreateAccountView__faqItemTextContainer columns small-12 medium-10 medium-offset-1 large-8 large-offset-0">
			<div className="Step1_CreateAccountView__faqItemLabel">{label}</div>
			<div className="Step1_CreateAccountView__faqItemDescription">{description}</div>
		</div>
	</div>
);

/**
 * A Functional React component for rendering the Browser Create Account View
 * @return {JSX} JSX for rendering the Browser Create Account View of the Hub app
 * @memberof GhosteryBrowserHubViews
 */
const Step1_CreateAccountView = (props) => {
	const { user, actions } = props;
	const { setSetupStep } = actions;
	const email = user && user.email;

	const [expanded, setExpanded] = useState(false);
	const [view, setView] = useState(CREATE_ACCOUNT);

	const arrowClassNames = ClassNames('Step1_CreateAccountView__arrow', {
		up: expanded,
		down: !expanded,
	});

	const faqRef = useRef(null);
	const scrollToFAQ = () => {
		faqRef.current.scrollIntoView({ behavior: 'smooth' });
	};

	const handleFAQLearnMoreClick = () => {
		setTimeout(scrollToFAQ, 1);
		setExpanded(!expanded);
	};

	const renderSkipLink = () => (
		<div className="row align-center-middle">
			<div className="columns small-10 medium-6" />
			<div className="columns small-10 medium-6">
				<NavLink className="Step1_CreateAccountView__skip" to="/onboarding/2" onClick={() => setSetupStep({ setup_step: LOGIN, origin: ONBOARDING })}>
					<span>{t('ghostery_browser_hub_onboarding_skip')}</span>
				</NavLink>
			</div>
		</div>
	);

	return (user ? (
		<div className="Step1_CreateAccountView__alreadySignedIn">
			<div className="Step1_CreateAccountView__title">{t('ghostery_browser_hub_onboarding_you_are_signed_in_as')}</div>
			<div className="Step1_CreateAccountView__email">{email}</div>
			<div className="Step1_CreateAccountView__ctaButtonContainer">
				<NavLink className="Step1_CreateAccountView__ctaButton" to="/onboarding/2" onClick={() => setSetupStep({ setup_step: LOGIN, origin: ONBOARDING })}>
					<span>{t('next')}</span>
				</NavLink>
			</div>
		</div>
	) : (
		<div className="Step1_CreateAccountView">
			{view === CREATE_ACCOUNT && (
				<div className="Step1_CreateAccountView__title">{t('ghostery_browser_hub_onboarding_create_a_ghostery_account')}</div>
			)}
			{view === SIGN_IN && (
				<div className="Step1_CreateAccountView__title">{t('sign_in')}</div>
			)}
			<div className="Step1_CreateAccountView__subtitle">{ t('ghostery_browser_hub_onboarding_sync_settings') }</div>
			<div className="row align-center-middle">
				{view === CREATE_ACCOUNT && (
					<div className="Step1_CreateAccountView__alreadyHaveAccount columns small-12 medium-6" onClick={() => setView(SIGN_IN)}>{t('ghostery_browser_hub_onboarding_already_have_account')}</div>
				)}
				{view === SIGN_IN && (
					<div className="Step1_CreateAccountView__alreadyHaveAccount columns small-12 medium-6" onClick={() => setView(CREATE_ACCOUNT)}>{t('ghostery_browser_hub_onboarding_create_an_account')}</div>
				)}
				<div className="Step1_CreateAccountView__alreadyHaveAccount columns small-12 medium-6" />
			</div>
			{view === CREATE_ACCOUNT ? (
				<Fragment>
					{/* eslint-disable-next-line react/jsx-pascal-case */}
					<Step1_CreateAccountForm />
					{renderSkipLink()}
					<div className="Step1_CreateAccountView__learnMoreContainer" onClick={handleFAQLearnMoreClick}>
						<div className="Step1_CreateAccountView__learnMore">{t('ghostery_browser_hub_onboarding_we_take_your_privacy_very_seriously')}</div>
					</div>
					<div className={arrowClassNames} onClick={handleFAQLearnMoreClick} />
					<div ref={faqRef} className="Step1_CreateAccountView__FAQContainer">
						{expanded &&
							faqList.map(item => renderFAQListItem(item.icon, item.label, item.description))
						}
					</div>
					{expanded && (
						<div className="row">
							<a className="Step1_CreateAccountView__privacyPolicyLink columns small-12 medium-10 medium-offset-1 large-8 large-offset-3" href={`${globals.GHOSTERY_BASE_URL}/about-ghostery/ghostery-plans-and-products-privacy-policy/`} target="_blank" rel="noreferrer">
								{t('ghostery_browser_hub_onboarding_visit_our_privacy_policy')}
							</a>
						</div>
					)}
				</Fragment>
			) : (
				<Fragment>
					{/* eslint-disable-next-line react/jsx-pascal-case */}
					<Step1_LogInForm />
					{renderSkipLink()}
				</Fragment>
			)}
		</div>
	));
};

// PropTypes ensure we pass required props of the correct type
Step1_CreateAccountView.propTypes = {
	user: PropTypes.shape({
		email: PropTypes.string,
	}),
};

// Default props used in the Plus View
Step1_CreateAccountView.defaultProps = {
	user: {
		email: ''
	},
};

export default Step1_CreateAccountView;
