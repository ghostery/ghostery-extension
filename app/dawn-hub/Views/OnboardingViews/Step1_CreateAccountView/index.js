/**
 * Point of entry index.js file for Dawn Hub onboarding flow Create Account View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { buildReduxHOC } from '../../../../shared-hub/utils';
import Step1_CreateAccountView from './Step1_CreateAccountView';
import { setSetupStep } from '../../../../shared-hub/actions/SetupLifecycleActions';
import setToast from '../../../../shared-hub/actions/ToastActions';

const actionCreators = {
	setSetupStep,
	setToast
};

export default buildReduxHOC(['account'], actionCreators, Step1_CreateAccountView);
