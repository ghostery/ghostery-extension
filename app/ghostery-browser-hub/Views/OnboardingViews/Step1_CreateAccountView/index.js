/**
 * Point of entry index.js file for Ghostery Browser Hub Create Account View
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2020 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import { withRouter } from 'react-router-dom';
import { buildReduxHOC } from '../../../../shared-hub/utils';
import Step1_CreateAccountView from './Step1_CreateAccountView';
import { setSetupStep } from '../../../../shared-hub/actions/SetupLifecycleActions';

const actionCreators = {
	setSetupStep,
};

export default withRouter(buildReduxHOC(['account'], actionCreators, Step1_CreateAccountView));
