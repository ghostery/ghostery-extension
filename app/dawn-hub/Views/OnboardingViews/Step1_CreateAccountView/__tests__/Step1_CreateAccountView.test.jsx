/**
 * Create Account View Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Step1_CreateAccountView from '../Step1_CreateAccountView';

jest.mock('../../Step1_CreateAccountForm', () => {
  const CreateAccountForm = () => <div />;
  return CreateAccountForm;
});

jest.mock('../../Step1_LogInForm', () => {
  const LogInForm = () => <div />;
  return LogInForm;
});

const noop = () => {};
describe('app/hub/Views/Step1_CreateAccountView component', () => {
	const initialState = {
		user: null,
		actions: {
			setSetupStep: noop
		}
	};
	describe('Snapshot tests with react-test-renderer', () => {
		test('Create Account Form view is rendered correctly', () => {
			const component = renderer.create(
				<MemoryRouter>
					<Step1_CreateAccountView {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('Create Account Form view is rendered correctly when user is logged in', () => {
			const plusUserState = {
				...initialState,
				user: {
					plusAccess: true
				}
			};

			const component = shallow(<Step1_CreateAccountView {...initialState} />);
		});
	});
});
