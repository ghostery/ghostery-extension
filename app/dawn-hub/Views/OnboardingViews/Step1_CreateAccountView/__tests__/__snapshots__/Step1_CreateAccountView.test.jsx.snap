// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/hub/Views/Step1_CreateAccountView component Snapshot tests with react-test-renderer Create Account Form view is rendered correctly 1`] = `
<div
  className="Step1_CreateAccountView"
>
  <div
    className="CreateAccountView__relativeContainer"
  >
    <div
      className="CreateAccountView__backContainer"
    >
      <span
        className="CreateAccountView__caret left"
      />
      <a
        aria-current={null}
        href="/onboarding/3"
        onClick={[Function]}
      >
        <span
          className="CreateAccountView__back"
        >
          ghostery_dawn_onboarding_back_to_search_selection
        </span>
      </a>
    </div>
  </div>
  <div
    className="Step1_CreateAccountView__title"
  >
    ghostery_dawn_onboarding_create_a_ghostery_account
  </div>
  <div
    className="Step1_CreateAccountView__subtitle"
  >
    ghostery_dawn_onboarding_create_account_for_trial
  </div>
  <div
    className="row align-center-middle"
  >
    <div
      className="Step1_CreateAccountView__alreadyHaveAccount columns small-12"
      onClick={[Function]}
    >
      ghostery_dawn_onboarding_already_have_account
    </div>
  </div>
  <div />
  <div
    className="Step1_CreateAccountView__FAQContainer"
  >
    <div
      className="Step1_CreateAccountView__faqItemContainer row"
    >
      <div
        className="Step1_CreateAccountView__faqIconContainer columns small-12 medium-10 large-2"
      >
        <img
          className="Step1_CreateAccountView__faqIcon"
          src="/app/images/hub/browser-create-account-view/ghosty-shield.svg"
        />
      </div>
      <div
        className="Step1_CreateAccountView__faqItemTextContainer columns small-12 medium-10 large-10"
      >
        <div
          className="Step1_CreateAccountView__faqItemLabel"
        >
          ghostery_dawn_onboarding_private_by_design
        </div>
        <div
          className="Step1_CreateAccountView__faqItemDescription"
        >
          ghostery_dawn_onboarding_private_by_design_description
        </div>
      </div>
    </div>
    <div
      className="Step1_CreateAccountView__faqItemContainer row"
    >
      <div
        className="Step1_CreateAccountView__faqIconContainer columns small-12 medium-10 large-2"
      >
        <img
          className="Step1_CreateAccountView__faqIcon"
          src="/app/images/hub/browser-create-account-view/ghosty-box.svg"
        />
      </div>
      <div
        className="Step1_CreateAccountView__faqItemTextContainer columns small-12 medium-10 large-10"
      >
        <div
          className="Step1_CreateAccountView__faqItemLabel"
        >
          ghostery_dawn_onboarding_can_i_remove_my_account
        </div>
        <div
          className="Step1_CreateAccountView__faqItemDescription"
        >
          ghostery_dawn_onboarding_can_i_remove_my_account_description
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <a
      className="Step1_CreateAccountView__privacyPolicyLink columns small-12 medium-10 medium-offset-1 large-8 large-offset-3"
      href="https://ghosterystage.com/about-ghostery/ghostery-plans-and-products-privacy-policy/"
      rel="noreferrer"
      target="_blank"
    >
      ghostery_dawn_onboarding_visit_our_privacy_policy
    </a>
  </div>
</div>
`;
